### PR TITLE
Revisions and restore functions.

### PR DESCRIPTION
**Pull request by elaquentil**
Les deux fonctions sont liées: utiliser les rev reçus de la fonction revision afin d'utiliser la fonction restore.

### CODE REVIEW

File: src/dropbox.mli [hunk 1]
```diff
@@ -276,6 +276,9 @@ module type S = sig
           contained in this folder. Return nothing if the folder is empty. *)
     }
 
+  type revisions = metadata list (** The list of the metadata for the
```
Comment (Chris00) [inline src/dropbox.mli]: Not sure it is useful to declare this type.  Inlining it would be OK.

File: src/dropbox.ml [hunk 2]
```diff
@@ -183,6 +185,8 @@ module type S = sig
       contents: metadata list;
     }
 
+  type revisions = metadata list
```
Comment (Chris00) [inline src/dropbox.ml]: See below — on Github ;-)

File: src/dropbox.atd [hunk 3]
```diff
@@ -64,3 +64,5 @@ type metadata = { size: string;
                   ?client_mtime: date option;
                   root: root;
                   ~contents: metadata list }
+
+type revisions = metadata list
```
Comment (Chris00) [inline src/dropbox.atd]: This is needed for the functions generation but it does not mean it must be exported.

**Comment by elaquentil** [comment]
I created a metadata_list type to be used by restore function and search function on the next commit.

**Comment by Chris00** [comment]
PR https://github.com/Chris00/ocaml-dropbox/pull/25 merged.

**Pull request closed without merging**